packages/shortcuts-extension/src/index.ts

```diff
@@ -161,8 +161,11 @@ const shortcuts: JupyterLabPlugin<void> = {
     }
 
     registry.pluginChanged.connect((sender, plugin) => {
-      if (!(plugin in loaded)) {
-        populate(canonical);
+      // If some other plugin changed, clear the shortcut cache and reload
+      // myself (repopulating the keyboard shortcuts)
+      if (plugin !== shortcuts.id) {
```

Comment:
The premise I was assuming was that defaults of a plugin won't change even if its user preferences change. Do you think that might not be the case? This will cause keyboard shortcuts to reload after *every* trivial user preference update and I think that might be overkill.

Reply:
Specifically, once a plugin has been loaded (and been transformed), I don't think its `jupyter.lab.shortcuts` will ever change again unless the author has written a `Transform` that is non-deterministically modifying the `jupyter.lab.shortcuts`, which is possible but seems both implausible and like bad behavior to me.

Reply:
That's (a) exactly what I was doing in my play example, and (b) totally possible in a transform, and indeed exactly what the shortcut extension does itself (dynamically change its keyboard shortcuts). If we have this assumption that no one else but the shortcuts extension does this, we should definitely document it somewhere prominent where people who author plugin settings will see it.

Reply:
On the other hand, isn't calculating the keyboard shortcuts relatively quick? I imagined that it wouldn't add too much overhead to make the behavior technically correct.